app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java

```diff
@@ -89,55 +92,31 @@ private static void addEvents(Context c, Uri uri) {
             }
         }
 
-        CalendarManager calendarManager = new CalendarManager(c);
-
-        // Get all calendar items from database
-        try (Cursor cursor = calendarManager.getAllFromDb()) {
-            while (cursor.moveToNext()) {
-                // Get each table row
-                //final String status = cursor.getString(1);
-                final String title = cursor.getString(3);
-                final String description = cursor.getString(4);
-                final String strStart = cursor.getString(5);
-                final String strEnd = cursor.getString(6);
-                final String location = cursor.getString(7);
-
-                // Get start and end time
-                long startMillis = Utils.getDateTime(strStart).getTime();
-                long endMillis = Utils.getDateTime(strEnd).getTime();
-
-                ContentValues values = new ContentValues();
-
-                // Put the received values into a contentResolver to
-                // transmit the to Google Calendar
-                values.put(CalendarContract.Events.DTSTART, startMillis);
-                values.put(CalendarContract.Events.DTEND, endMillis);
-                values.put(CalendarContract.Events.TITLE, title);
-                values.put(CalendarContract.Events.DESCRIPTION, description);
-                values.put(CalendarContract.Events.CALENDAR_ID, id);
-                values.put(CalendarContract.Events.EVENT_LOCATION, location);
-                values.put(CalendarContract.Events.EVENT_TIMEZONE, R.string.calendarTimeZone);
-                contentResolver.insert(CalendarContract.Events.CONTENT_URI, values);
-            }
+        CalendarDao calendarDao = TcaDb.getInstance(c).calendarDao();
+
+        List<CalendarItem> calendarItems = calendarDao.getAllNotCancelled();
+        for (CalendarItem calendarItem: calendarItems) {
+            ContentValues values = new ContentValues();
+
+            // Put the received values into a contentResolver to
+            // transmit the to Google Calendar
+            values.put(CalendarContract.Events.DTSTART, Utils.getDateTime(calendarItem.getDtstart()).getTime());
+            values.put(CalendarContract.Events.DTEND, Utils.getDateTime(calendarItem.getDtend()).getTime());
+            values.put(CalendarContract.Events.TITLE, calendarItem.getTitle());
+            values.put(CalendarContract.Events.DESCRIPTION, calendarItem.getDescription());
+            values.put(CalendarContract.Events.CALENDAR_ID, id);
+            values.put(CalendarContract.Events.EVENT_LOCATION, calendarItem.getLocation());
+            values.put(CalendarContract.Events.EVENT_TIMEZONE, R.string.calendarTimeZone);
```

Comment:
This should probably be a method: `CalendarItem.toContentValues()`

Reply:
Will leave out CALENDAR_ID and EVENT_TIMEZONE as they seem to be out of CalendarItem's scope